### PR TITLE
Drop contiguity info for size 1 dimensions for Python's define_tensor(sizes=..., strides=...)

### LINKED ISSUE #2549: broadcast_in_dim: The size of contiguity must equal to the number of non-broadcasting IterDomains
### 🐛 Describe the bug

```py
from nvfuser import FusionDefinition, DataType
import torch

def nvfuser_fusion_id2(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(symbolic_sizes=[-1, 1], contiguous=[True, True], dtype=DataType.Double, is_cpu=False)
    T1 = fd.define_tensor(symbolic_sizes=[-1, -1], contiguous=[True, True], dtype=DataType.Double, is_cpu=False)
    T2 = fd.ops.broadcast_in_dim(T0, output_shape=[4, 4], broadcast_dims=[0, 1])
    T3 = fd.ops.div(T1, T2)
    fd.add_output(T3)

a = torch.ones(4, 1, dtype=torch.double, device='cuda')
b = torch.ones(4, 4, dtype=torch.double, device='cuda')

# RuntimeError: The size of contiguity must equal to the number of non-broadcasting IterDomains
with FusionDefinition() as fd:
    nvfuser_fusion_id2(fd)

fd.execute([a, b])
```

### Versions

devel

### PR DESCRIPTION
**Pull request by IvanYashchuk**
Fixes https://github.com/csarofeen/pytorch/issues/2549.

### CODE REVIEW

File: third_party/nvfuser/csrc/python_frontend/python_bindings.cpp
Comment (zasdfgbnm) [inline third_party/nvfuser/csrc/python_frontend/python_bindings.cpp]: Strange. Isn't this already handled by `computeContiguity`?https://github.com/csarofeen/pytorch/blob/976b43c41da9e32c735beb775f22cdb5521b3c90/third_party/nvfuser/csrc/python_frontend/python_bindings.cpp#L28

Comment (IvanYashchuk) [reply third_party/nvfuser/csrc/python_frontend/python_bindings.cpp]: Here, in `maybe_symbolic_sizes`, there's a specialization on size 1 dimensions independent of strides, while `computeContiguity` checks stride==0 for the dimension to be considered as the broadcast dimension.

Maybe that line should be changed to `auto not_broadcast = [&](auto i) { return sizes[i] != 1; }; `?

Comment (zasdfgbnm) [reply third_party/nvfuser/csrc/python_frontend/python_bindings.cpp]: Ahh, I see. I am OK with changing `not_broadcast` for now to unblock. But I think a better solution is to a some refactor on the python frontend to handle size N stride 0 dimension as expanded broadcast.

**Review by zasdfgbnm — APPROVED** [review]
Thanks for fixing!

File: third_party/nvfuser/csrc/python_frontend/python_bindings.cpp
Comment (zasdfgbnm) [inline third_party/nvfuser/csrc/python_frontend/python_bindings.cpp:29]: Probably this should be `sizes[i] != 1`? `test_compute_contiguity` might fail if we do so, but we can skip that test for now.

**Comment by jjsjann123** [comment]
Not trying to block this PR.

think about a tensor `torch.randn(5, 5).expand(-1).expand((5, 5, 5))`

We still have issue with explicitly broadcast dimensions that I don't think is handled by this PR yet. With the API on TensorOpRecord, we can't properly pass correct contiguity information for an expanded dimension.
I'm trying to keep the contiguity information more comprehensible on the python side (not skipping contiguity for broadcast dims).

Wondering if you want to patch that in this PR as well? otherwise I'll start one once this merges.
cc'ing @IvanYashchuk

File: third_party/nvfuser/csrc/python_frontend/python_bindings.cpp
Comment (jjsjann123) [reply third_party/nvfuser/csrc/python_frontend/python_bindings.cpp]: #2561 handles size N stride 0. Should we move along with that instead of this PR?

Comment (IvanYashchuk) [reply third_party/nvfuser/csrc/python_frontend/python_bindings.cpp]: Yes, let's move with https://github.com/csarofeen/pytorch/pull/2561!

**Comment by IvanYashchuk** [comment]
Closing in favor of https://github.com/csarofeen/pytorch/pull/2561

**Pull request closed without merging**
